pkg/disttask/framework/storage/task_table.go

```diff
@@ -690,14 +690,15 @@ func (stm *TaskManager) StartManager(ctx context.Context, tidbID string, role st
 }
 
 // StartManagerSession insert the manager information into dist_framework_meta.
-// if the record exists, update the cpu_count.
+// if the record exists, update the cpu_count and role.
 func (*TaskManager) StartManagerSession(ctx context.Context, se sessionctx.Context, execID string, role string) error {
 	cpuCount := cpu.GetCPUCount()
 	_, err := sqlexec.ExecSQL(ctx, se, `
 		insert into mysql.dist_framework_meta(host, role, cpu_count, keyspace_id)
 		values (%?, %?, %?, -1)
-		on duplicate key update cpu_count = %?`,
-		execID, role, cpuCount, cpuCount)
+		on duplicate key
+		update cpu_count = %?, role = %?`,
```

Comment:
we'll call it periodically in taskexecutor manager, will override it

we need another interface to change/update role

Reply:
Won't override it, since we have store the role in global memory

Reply:
If the tidb crash or exit, it's safe to override it.

Reply:
suppose we `set` before `initMeta` buf after get get `config.GetGlobalConfig().Instance.TiDBServiceScope`, the scope will be wrong for 90s

Reply:
There are 2 timeflow:
1. 
a. config.GetGlobalConfig().Instance.TiDBServiceScope = valStr
b. startManager in recoverLoop
c. startManager in set.go

2. 
a. config.GetGlobalConfig().Instance.TiDBServiceScope = valStr
b. startManager in set.go
c. startManager in recoverLoop


both of them will use the latest one.

Reply:
If the above scenario can be written a case to cover it？